### PR TITLE
allow diff for non staged files

### PR DESCRIPTION
**Pull request by ermirizio**

### CODE REVIEW

File: .github/workflows/doctor.yml [hunk 1]
```diff
@@ -28,4 +28,8 @@ jobs:
 
       - name: Check for zero diffs
         id: diff
-        run: gromit git diff ${{ matrix.repo }}
+        run: |
+          output=$(gromit git --loglevel trace diff ${{ matrix.repo }} 2>&1)
```
Comment (alephnull) [inline .github/workflows/doctor.yml:32]: Is trace level needed here?

File: git/diff.go [hunk 2]
```diff
@@ -9,9 +9,14 @@ import (
 	"github.com/waigani/diffparser"
 )
 
-func gitDiff(dir string) (string, error) {
+func gitDiff(dir string, isPR bool) (string, error) {
 	var out bytes.Buffer
-	cmd := exec.Command("git", "diff", "-G", "(^[^#])", "--staged")
+	var cmd *exec.Cmd
+	if isPR {
```
Comment (alephnull) [inline git/diff.go:15]: isPR doesn't seem like the correct condition. `gromit git diff` has no connection to PRs. I think it makes more sense to combine the staged and unstaged diffs into one.

**Review by alephnull — CHANGES_REQUESTED** [review]
Thanks for this! But I think the implementation needs a bit more work.

File: git/diff.go (hunk 2 above)
Comment (alephnull) [reply git/diff.go:15]: You need a named commit, which I've implemented in my pr.

**Comment by alephnull** [comment]
I think this can be abandoned.

File: .github/workflows/doctor.yml (hunk 1 above)
Comment (ermirizio) [reply .github/workflows/doctor.yml:32]: Not really, I would output this to the info filter.

**Pull request closed without merging**